### PR TITLE
Configuration UX

### PR DESCRIPTION
**Pull request by jsoverson**
This PR:
- Starts to address #34
- Adds errors when nanobus encounters unknown fields
- Adds errors when fields are unset when they should have values.
- Adds defaults to processed yaml to avoid errors arising from above.

A lot of the big looking changes are either newly generated files or moving some code around to avoid cycles and/or centralize error handling.

### CODE REVIEW

**Comment by jsoverson** [comment]
Just noticed tests fail after I updated the retry decoding. The `DecodeConfig` tests seem to test a configuration that `DecodeConfig` should never have to decode. The only reference I can see to `DecodeConfig` passes it a `Backoff` struct which is formatted differently. Is it OK to change the tests and base them off `Backoff` vs the arbitrary config maps?

File: pkg/codec/cloudevents/json/json.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 type (
 	Config struct {
 		SpecVersion string `mapstructure:"specversion"`
-		Source      string `mapstructre:"source"`
+		Source      string `mapstructure:"source" structs:"-"`
```
Comment (pkedy) [inline pkg/codec/cloudevents/json/json.go:26]: `structs:"-"`

What is this for?

File: pkg/config/decode.go [hunk 2]
```diff
@@ -48,10 +48,11 @@ type FromStringer interface {
 //
 // Most of the heavy lifting is handled by the mapstructure library. A custom decoder is used to handle
 // decoding string values to the supported primitives.
-func Decode(input interface{}, output interface{}) error {
+func Decode(input interface{}, output interface{}, defaults ...interface{}) error {
```
Comment (pkedy) [inline pkg/config/decode.go:51]: `defaults ...interface{}` can be removed now, correct?

**Review by pkedy — COMMENTED** [review]
Just a few questions

File: pkg/codec/cloudevents/json/json.go (hunk 1 above)
Comment (jsoverson) [reply pkg/codec/cloudevents/json/json.go]: It was for the old validator. Didn't notice that didn't get removed. It's gone now.

File: pkg/config/decode.go (hunk 2 above)
Comment (jsoverson) [reply pkg/config/decode.go]: Yep, it's gone now.

**Review by pkedy — APPROVED** [review]
LGTM

**Pull request merged**